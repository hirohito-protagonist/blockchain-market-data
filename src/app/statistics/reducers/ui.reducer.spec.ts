import { fromUIActions } from './../actions/index.action';
import { reducer, UIState } from './ui.reducer';
import { reduce } from 'rxjs/operators';


describe('UI reducer store state', () => {

  describe('undefiend action', () => {

    it('should return default state', () => {

      const action = { type: void(0) } as fromUIActions.ActionType;
      const result = reducer(void(0), action);

      expect(result).toMatchSnapshot();
    });
  });

  describe('[Statistics UI] update ui state', () => {

    it('should not add new property to the store', () => {

      const action = fromUIActions.updateUIState({
        payload: { key: 'test', value: 'test' }
      });
      const result = reducer(void(0), action);

      expect(result).toMatchSnapshot();
    });

    it('should partial update state for provided key', () => {

      const value1 = { selectedChart: 'test' };
      let action = fromUIActions.updateUIState({
        payload: { key: 'chartsView', value: value1 }
      });
      let result = reducer(void(0), action);

      expect(result).toMatchSnapshot();

      const value2 = { selectedChartTimeSpan: '20days' };
      action = fromUIActions.updateUIState({
        payload: { key: 'chartsView', value: value2 }
      });
      result = reducer(result, action);

      expect(result).toMatchSnapshot();
    });
  });
});
