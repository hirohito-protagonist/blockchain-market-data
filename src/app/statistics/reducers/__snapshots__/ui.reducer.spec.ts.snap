// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UI reducer store state [Statistics UI] update ui state should not add new property to the store 1`] = `
Object {
  "chartsView": Object {
    "selectedChart": "transactions-per-second",
    "selectedChartTimeSpan": "30days",
  },
}
`;

exports[`UI reducer store state [Statistics UI] update ui state should partial update state for provided key 1`] = `
Object {
  "chartsView": Object {
    "selectedChart": "test",
    "selectedChartTimeSpan": "30days",
  },
}
`;

exports[`UI reducer store state [Statistics UI] update ui state should partial update state for provided key 2`] = `
Object {
  "chartsView": Object {
    "selectedChart": "test",
    "selectedChartTimeSpan": "20days",
  },
}
`;

exports[`UI reducer store state undefiend action should return default state 1`] = `
Object {
  "chartsView": Object {
    "selectedChart": "transactions-per-second",
    "selectedChartTimeSpan": "30days",
  },
}
`;
